src/RoomsSDKAdapter.js

```diff
@@ -134,4 +153,138 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
 
     return this.getRoomObservables[ID];
   }
+
+  /**
+   * Returns an array of IDs of the most recent activities in a conversation up to the specified limit.
+   *
+   * @param {object} Params.ID ID for the room
+   * @param {object} Params.earliestActivityDate  Get all child activities before this date
+   * @returns {Promise} Resolves with array of activities
+   * @private
+   */
+  async fetchActivities({
+    ID,
+    earliestActivityDate,
+  }) {
+    const {limit} = this;
+    const conversationId = deconstructHydraId(ID).id;
+
+    logger.debug('ROOM', 'fetchActivities()', ['called with', {
+      ID,
+      limit,
+      earliestActivityDate,
+    }]);
+
+    if (!FETCHED_CONVERSATIONS) {
+      await this.datasource.internal.conversation.list();
+      FETCHED_CONVERSATIONS = true;
+    }
+
+    return this.datasource.internal.conversation.listActivities({
+      conversationId,
+      limit: limit + 1,
+      lastActivityFirst: true,
+      maxDate: earliestActivityDate === null ? undefined : earliestActivityDate,
+    });
+  }
+
+  /**
+   * Returns `true` if there are more activities to load from the room of the given ID.
+   * Otherwise, it returns `false`.
+   *
+   * @param {string} ID ID of the room for which to verify activities.
+   * @returns {boolean} `true` if room has more activities to load, `false` otherwise
+   */
+  hasMoreActivities(ID) {
+    const roomObservable = this.roomObservables[ID];
+    const {
+      hasMore = true,
+    } = this.roomActivities.get(ID);
+
+    if (!hasMore) {
+      roomObservable.complete();
+    } else {
+      this.fetchPastActivities(ID);
+    }
+
+    return hasMore;
+  }
+
+  /**
+   * Fetches past activities and returns array of (id, published) objects.
+   *
+   * @param {string} ID The id of the room
+   * @returns {Subscription} Observable stream that emits activity data
+   */
+  fetchPastActivities(ID) {
+    const roomActivity = this.roomActivities.get(ID);
+    const {earliestActivityDate} = roomActivity;
+    const roomObservable = this.roomObservables[ID];
+
+    logger.debug('ROOM', 'fetchPastActivities()', ['called with', {
+      ID,
+      earliestActivityDate,
+    }]);
+
+    if (!ID) {
+      roomObservable.error(new Error('getPreviousActivities - Must provide room ID'));
+    }
+
+    return from(this.fetchActivities({
+      ID,
+      earliestActivityDate,
+    }))
+      .subscribe((data) => {
```

Comment:
I don't think this needs to be an observable, but it is also not a negative thing that it is. The only thing I would change is that if you're going to go the observable route, keep it an observable instead of making it a subscriptions, they are not the same things and cannot be chained